tests/sentry/backup/test_correctness.py

```diff
@@ -0,0 +1,188 @@
+
+from dateutil import parser
+from django.db import IntegrityError
+from jsondiff import diff
+from pydantic import FilePath, PositiveInt
+
+from sentry.db.postgres.roles import in_test_psql_role_override
+from sentry.runner.commands.backup import exporter, importer
+from sentry.utils import json
+from sentry.utils.json import JSONData
+from sentry.utils.pytest.fixtures import django_db_all
+
+ComparatorName = NewType("ComparatorName", str)
+ModelName = NewType("ModelName", str)
+
+
+class InstanceID:
+    """Every entry in the generated backup JSON file should have a unique model+pk combination, which serves as its identifier."""
+
+    model: ModelName
+    pk: PositiveInt
+
+    def __init__(self, model: ModelName, pk: PositiveInt):
+        self.model = model
+        self.pk = pk
+
+    def __eq__(self, other):
+        return (
+            hasattr(other, "model")
+            and self.model == other.model
+            and hasattr(other, "pk")
+            and self.pk == other.pk
+        )
+
+    def __hash__(self):
+        return hash((self.model, self.pk))
+
+    def __str__(self):
+        return f'InstanceID(model: "{self.model}", pk: {self.pk})'
+
+
+class ComparatorFinding:
+    """Store all information about a single failed matching between expected and actual output."""
+
+    name: ComparatorName
+    on: InstanceID
+    reason: str
+
+    def __init__(self, name: ComparatorName, on: InstanceID, reason: Optional[str] = ""):
+        self.name = name
+        self.on = on
+        self.reason = reason
+
+    def __str__(self):
+        return (
+            f'Finding(\n\tname: "{self.name}",\n\ton: {str(self.on)},\n\treason: {self.reason}\n)'
+        )
+
+
+class ComparatorFindings(List[ComparatorFinding]):
+    """A wrapper type for a list of 'ComparatorFinding' which enables pretty-printing in asserts."""
+
+    def __str__(self):
+        out = ""
+        for finding in self:
+            out += "\n" + str(finding)
+        return out
+
+    def maybe_assert(self):
+        if len(self) > 0:
+            assert False, str(self)
+
+
+class JSONMutatingComparator(Protocol):
+    """A callback protocol specifying the function signature for custom JSON comparators."""
+
+    def __call__(self, on: InstanceID, expect: JSONData, actual: JSONData) -> Optional[str]:
+        ...
+
+
+def comparator_date_updated(on, expect, actual):
+    exp_date_updated = parser.parse(expect["fields"]["date_updated"])
+    act_date_updated = parser.parse(actual["fields"]["date_updated"])
+    if not act_date_updated >= exp_date_updated:
+        return f"{expect['fields']['date_updated']} was not >= {actual['fields']['date_updated']}"
+    else:
+        expect["fields"]["date_updated"] = "__COMPARATOR_DATE_UPDATED__"
+        actual["fields"]["date_updated"] = "__COMPARATOR_DATE_UPDATED__"
+
+
+REPO_PATH = Path(os.path.dirname(os.path.realpath("__file__")))
+SNAPSHOT_PATH = REPO_PATH / "tests/sentry/backup/snapshots"
```

Comment:
there's a `fixtures` directory intended for these

Reply:
Done.